custom_components/tahoma/binary_sensor.py

```diff
@@ -1,64 +1,94 @@
 """Support for TaHoma binary sensors."""
-from typing import Optional
 
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Callable
+
+from homeassistant.components import binary_sensor
 from homeassistant.components.binary_sensor import (
-    DEVICE_CLASS_MOTION,
-    DEVICE_CLASS_OCCUPANCY,
-    DEVICE_CLASS_OPENING,
-    DEVICE_CLASS_SMOKE,
-    DOMAIN as BINARY_SENSOR,
     BinarySensorEntity,
+    BinarySensorEntityDescription,
 )
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
 from .const import DOMAIN
+from .coordinator import TahomaDataUpdateCoordinator
 from .tahoma_entity import TahomaEntity
 
-CORE_ASSEMBLY_STATE = "core:AssemblyState"
-CORE_BUTTON_STATE = "core:ButtonState"
-CORE_CONTACT_STATE = "core:ContactState"
-CORE_GAS_DETECTION_STATE = "core:GasDetectionState"
-CORE_OCCUPANCY_STATE = "core:OccupancyState"
-CORE_OPENING_STATE = "core:OpeningState"
-CORE_OPEN_CLOSED_TILT_STATE = "core:OpenClosedTiltState"
-CORE_RAIN_STATE = "core:RainState"
-CORE_SMOKE_STATE = "core:SmokeState"
-CORE_THREE_WAY_HANDLE_DIRECTION_STATE = "core:ThreeWayHandleDirectionState"
-CORE_VIBRATION_STATE = "core:VibrationState"
-CORE_WATER_DETECTION_STATE = "core:WaterDetectionState"
-
-DEVICE_CLASS_BUTTON = "button"
-DEVICE_CLASS_GAS = "gas"
-DEVICE_CLASS_RAIN = "rain"
-DEVICE_CLASS_WATER = "water"
-
-ICON_WATER = "mdi:water"
-ICON_WATER_OFF = "mdi:water-off"
-ICON_WAVES = "mdi:waves"
-ICON_WEATHER_RAINY = "mdi:weather-rainy"
-
-IO_VIBRATION_DETECTED_STATE = "io:VibrationDetectedState"
-
 STATE_OPEN = "open"
 STATE_PERSON_INSIDE = "personInside"
 STATE_DETECTED = "detected"
-STATE_PRESSED = "pressed"
-
-TAHOMA_BINARY_SENSOR_DEVICE_CLASSES = {
-    "AirFlowSensor": DEVICE_CLASS_GAS,
-    "CarButtonSensor": DEVICE_CLASS_BUTTON,
-    "ContactSensor": DEVICE_CLASS_OPENING,
-    "MotionSensor": DEVICE_CLASS_MOTION,
-    "OccupancySensor": DEVICE_CLASS_OCCUPANCY,
-    "RainSensor": DEVICE_CLASS_RAIN,
-    "SirenStatus": DEVICE_CLASS_OPENING,
-    "SmokeSensor": DEVICE_CLASS_SMOKE,
-    "WaterDetectionSensor": DEVICE_CLASS_WATER,
-    "WaterSensor": DEVICE_CLASS_WATER,
-    "WindowHandle": DEVICE_CLASS_OPENING,
-}
+
+
+@dataclass
+class OverkizBinarySensorDescription(BinarySensorEntityDescription):
+    """Class to describe a Overkiz binary sensor."""
+
+    is_on: Callable[[Any], Any] = lambda state: state
```

Comment:
State is already a boolean?

Reply:
What do you mean? We use `is_on` to do the transformation, where we check e.g. is the state 'open' (binary = on), otherwise binary = off.

Reply:
Sorry, in my mind, it should returned a boolean because of the suffix `is_`.

Reply:
The lambda will return a boolean? Could we type this in a better way?

The `is_on` is typed as Callable, since this is the lambda function that eventually will return a boolean.